app/graphql/types/profile.rb

```diff
@@ -32,19 +37,55 @@ class Types::Profile < Types::BaseObject
     null: true,
     description: 'The character this profile has declared as their waifu or husbando'
 
-  field :waifu_or_husbando, String,
+  field :waifu_or_husbando, Types::WaifuOrHusbando,
     null: true,
-    description: 'The user-provided (unsanitized) string used to identify the role of the waifu'
+    description:
+      <<~DESCRIPTION.strip
```

Comment:
There's basically two whitespace-stripping methods in Rails/Ruby (aside from the legacy `.strip_heredoc` which was replaced by `<<~` which you use here):

- String#strip removes whitespace at the beginning and end but leaves it in the middle, so you get this:
  ```
  "The user-provided (unsanitized) string used to identify\nthe role of the waifu. 'Waifu' or 'Husbando'"
  ```
- String#squish collapses all runs of more than 1 whitespace (including newlines) with a single space, giving this:
  ```ruby
  "The user-provided (unsanitized) string used to identify the role of the waifu. 'Waifu' or 'Husbando'"
  ```

You probably want `.squish` in this case!

That said, this should also be changed to reflect the use of an enum here instead of a user-provided string, and we should verify what GraphQL does when we return a legacy value here (does it just return null or does it crash?)

```suggestion
      <<~DESCRIPTION.squish
```